### PR TITLE
26841 Update Sample Bar Chart to support high contrast mode

### PR DESCRIPTION
**Pull request by ignatvilesov**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,4 +2,5 @@ typings
 node_modules
 .DS_Store
 .tmp
-dist
\ No newline at end of file
+dist
+.api
```
Comment (Guy-Moses) [inline .gitignore]: Thanks for cleanup and adding this

File: src/barChart.ts [hunk 2]
```diff
@@ -94,31 +99,36 @@ module powerbi.extensibility.visual {
...
+            const color: string = getColumnColorByIndex(category, i, colorPalette);
+
+            const selectionId: ISelectionId = host.createSelectionIdBuilder()
+                .withCategory(category, i)
+                .createSelectionId();
 
             barChartDataPoints.push({
-                category: category.values[i] + '',
+                color,
+                strokeColor,
```
Comment (Guy-Moses) [inline src/barChart.ts]: Please use thicker strokes for high-contrast mode (native visuals use `stroke-width 2`)

File: src/barChart.ts [hunk 3]
```diff
@@ -130,6 +140,57 @@ module powerbi.extensibility.visual {
...
+        ).solid.color;
+    }
+
+    function getColumnStrokeColor(colorPalette: ISandboxExtendedColorPalette): string {
+        return colorPalette.isHighContrast
+            ? colorPalette.foreground.value
+            : null;
+    }
+
+    function getAxisTextFillColor(
```
Comment (Guy-Moses) [inline src/barChart.ts]: I can't spot the bug here, but when testing the visual, axis text is always black, it should be foreground color.

File: src/barChart.ts [hunk 4]
```diff
@@ -260,9 +325,12 @@ module powerbi.extensibility.visual {
                     height: d => height - yScale(<number>d.value),
                     y: d => yScale(<number>d.value),
                     x: d => xScale(d.category),
-                    fill: d => d.color,
                 })
-                .style('fill-opacity', viewModel.settings.generalView.opacity / 100);
+                .style({
+                    'fill-opacity': viewModel.settings.generalView.opacity / 100,
```
Comment (Guy-Moses) [inline src/barChart.ts]: Please add stroke opacity for highlight indication in high-contrast mode (native use `stroke-opacity 0.4`)

File: .gitignore (hunk 1 above)
Comment (ignatvilesov) [reply .gitignore]: :smile:

File: src/barChart.ts (hunk 2 above)
Comment (ignatvilesov) [reply src/barChart.ts]: Done

File: src/barChart.ts (hunk 3 above)
Comment (ignatvilesov) [reply src/barChart.ts]: done. Issue was related to applying `color` instead of `fill`

File: src/barChart.ts (hunk 4 above)
Comment (ignatvilesov) [reply src/barChart.ts]: done. Thanks for your notes.

**Pull request merged**